spark/src/main/java/org/apache/iceberg/spark/source/Writer.java

```diff
@@ -425,17 +498,18 @@ public WriterCommitMessage commit() throws IOException {
 
     @Override
     public void abort() throws IOException {
+      closeCurrent();
+
       // clean up files created by this writer
       Tasks.foreach(completedFiles)
           .throwFailureWhenFinished()
           .noRetry()
           .run(file -> fileIo.deleteFile(file.path().toString()));
+    }
 
-      if (currentAppender != null) {
-        currentAppender.close();
-        this.currentAppender = null;
-        fileIo.deleteFile(currentFile.encryptingOutputFile());
-      }
+    private void openCurrent() {
```

Comment:
It would be nice to share more code between the partitioned and unpartitioned writers, like the open and close methods. Is it possible to refactor so that these are shared and the unpartitioned writer uses an empty key?

Reply:
You can detect an empty key by checking its size.

Reply:
I though about this, but was not sure that this PR should fix that. You ok if I do it as a follow up PR, or you'd rather have it taken care of here?

Reply:
Took a stab at the refactoring in this PR, but as its own individual commit so that its clear what has changed.